### PR TITLE
Email Authentication Supabase

### PR DESCRIPTION
**Pull request by mgkram4**
… form

**What is the new behavior that this PR brings (describe or link to an issue)?**

Email login with Supabase

The login Button will route the user to a Login page, with a login form 

**PR checklist (requester)**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Your code is documented

**PR checklist (reviewer)**

// TODO

**Other information**:

### CODE REVIEW

File: .env.local.example [hunk 1]
```diff
@@ -1,2 +1,5 @@
  NEXT_PUBLIC_SUPABASE_URL=your-project-url
- NEXT_PUBLIC_SUPABASE_ANON_KEY=your-project-key
\ No newline at end of file
+ NEXT_PUBLIC_SUPABASE_ANON_KEY=your-project-key
+ 
+
+ https://xnjgzwpzppkttqesxmhj.supabase.co
```
Comment (blbergo) [inline .env.local.example]: This is an environment variable and shouldn't be put into version control

File: api/supabase.ts [hunk 2]
```diff
@@ -2,3 +2,13 @@ import { createClient } from "@supabase/supabase-js";
 import { SUPABASE_KEY, SUPABASE_URL } from "@config/constants";
 
 export const supabase = createClient(SUPABASE_URL, SUPABASE_KEY);
+
+async function signUpNewUser() {
```
Comment (blbergo) [inline api/supabase.ts:7]: needs documentation

File: app/routes/login/page.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import SignUpForm from "@components/signup-form";
+
+export default function Login() {
+  return (
+    <div className="flex items-center justify-center min-h-screen bg-slate-100">
+      <SignUpForm />
```
Comment (blbergo) [inline app/routes/login/page.tsx]: The actual signup form component itself is redundant, if we aren't going to reuse it as a component, then it shouldn't get its own file

File: lib/components/navbar.tsx [hunk 4]
```diff
@@ -33,9 +34,11 @@ export default function Navbar() {
               </a>
             </li>
             <li>
-              <button className="block text-white  hover:bg-green-700 p-2 rounded-xl bg-green-600 ">
-                Login
-              </button>
+              <Link href="/routes/login">
```
Comment (blbergo) [inline lib/components/navbar.tsx]: we can code the routes into its own file with just an enum or an object that maps to each route

File: lib/components/signup-form.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+import { supabase } from "@api/supabase";
+import React, { useState } from "react";
+
+export default function SignUpForm() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  const signUpNewUser = async () => {
+    const { data, error } = await supabase.auth.signUp({
```
Comment (blbergo) [inline lib/components/signup-form.tsx]: later I'll go through and hook this up to our database

**Review by blbergo — CHANGES_REQUESTED** [review]
overall looks good, just checkout my comments

File: api/supabase.ts (hunk 2 above)
Comment (mgkram4) [reply api/supabase.ts:7]: could you explain this a little more

**Review by blbergo — APPROVED** [review]

**Pull request merged**